src/macaron/slsa_analyzer/git_url.py

```diff
@@ -376,6 +377,47 @@ def clone_remote_repo(clone_dir: str, url: str) -> Repo | None:
     return Repo(path=clone_dir)
 
 
+def list_remote_references(arguments: list[str], repo: str) -> str | None:
+    """Retrieve references from a remote repository using Git's ``ls-remote``.
+
+    Parameters
+    ----------
+    arguments: list[str]
+        The arguments to pass into the command.
+    repo: str
+        The repository to run the command on.
+
+    Returns
+    -------
+    str
+        The result of the command.
+    """
+    try:
+        result = subprocess.run(  # nosec B603
+            args=["git", "ls-remote"] + arguments + [repo],
+            capture_output=True,
+            # By setting stdin to /dev/null and using a new session, we prevent all possible user input prompts.
+            stdin=subprocess.DEVNULL,
+            start_new_session=True,
```

Comment:
Just to confirm: we will use this to resolve the issue instead of relying to patching the env variable `GIT_TERMINAL_PROMPT`.
This is a different approach to https://github.com/oracle/macaron/blob/c1c4eea36fafd35f72b9409c9d4df119cbb07451/src/macaron/slsa_analyzer/git_url.py#L351-L365

Reply:
It is a different approach. Possibly it is more robust as it prevents any requests for input instead of just a specific type, but it may well be that the specific type is the only one we need to worry about.